### ISSUE TITLE
Timezone conversion

### ISSUE THREAD

**Issue by StiventoUser** [matthijs/sqlpp11-connector-postgresql#46]
Hi matthijs,

Recently i noticed a problem with extraction timezone from the db.
1. Test generates datetime '2018-03-26 22:46:03.478427'
2. It's uploaded as '2018-03-26 22:46:03.478427-8:0'  ('2018-03-27 06:46:03.478427+0')
3. But the server returns '2018-03-26 23:46:03.478427-07' ('2018-03-27 06:46:03.478427+0')

They are same, but as I understand currently [timezone](https://github.com/matthijs/sqlpp11-connector-postgresql/blob/master/src/bind_result.cpp#L315) is ignored.
Have I missed something?

```
PostgreSQL debug: preparing: UPDATE tab_date_time SET col_day_point=$1,col_time_point=$2
...
---- running prepared update ----
...
PostgreSQL debug: binding date_time parameter string: 2018-03-26 22:46:03.478427-8:0
...
---- finished prepared update ----
PostgreSQL debug: executing: SELECT tab_date_time.col_day_point,tab_date_time.col_time_point FROM tab_date_time
...
PostgreSQL debug: binding date_time result at index: 1
PostgreSQL debug: got date_time string: 2018-03-26 23:46:03.478427-07
PostgreSQL debug: calculated timestamp 
900-01-01 00:00:00 CEST
141: TIMESTAMP '2018-03-26 23:46:03.478427' != TIMESTAMP '2018-03-26 22:46:03.478427'
```

**Comment by StiventoUser** [comment]
Also, could you please help me with the code?
Why it was commented?

**Comment by matthijs** [comment]
I've looked into this, but I didn't write that part of the library. As far as I can see this can use some refactoring using the date library. Parsing is added a while ago and is done in the following way:
```c++
std::istringstream in {datetime_iso_string};
date::micro_seconds tp;
is >> date::parse("%FT%T%z", tp);
```

See [here](https://howardhinnant.github.io/date/date.html#from_stream_formatting) for the formatting options.

**Comment by StiventoUser** [comment]
I think it can't be parsed with date::parse because I had a situation when the PostgreSQL returned an arbitrary number of digits in the fractional part.

I'll try to uncomment the lines to see if it help.

**Comment by matthijs** [comment]
According to [this](https://www.postgresql.org/docs/current/static/datatype-datetime.html) documentation, the resolution is 1 microsecond. Which datatype did you have that returned an arbitrary number of digits?

**Comment by StiventoUser** [comment]
Timestamp.
e.g. instead this date 2018-03-27 06:46:03.40000 it returned this: 2018-03-27 06:46:03.4

**Comment by StiventoUser** [comment]
Here is old CI status I've found: https://travis-ci.org/matthijs/sqlpp11-connector-postgresql/jobs/346028969

with the line 
`PostgreSQL debug: got date_time string: 2018-02-25 20:30:10.57572` 
and then
`TIMESTAMP '2018-02-25 20:30:10.000000' != TIMESTAMP '2018-02-25 20:30:10.575720'`

That's the reason why I've changed the ms parsing behavior in the last PR.

**Comment by StiventoUser** [comment]
Also, Am I correct that the library stores time in the UTC?

**Comment by matthijs** [comment]
The sqlpp library uses sqlpp::chrono::microsecond_point as time_point. And the microsecond_point is defined as follows:
```c++
using microsecond_point = std::chrono::time_point<std::chrono::system_clock, std::chrono::microseconds>;
```

According to the [system_clock](http://en.cppreference.com/w/cpp/chrono/system_clock) documentation it is stored as UTC. So yes, you are correct about storing the time in UTC.

**Comment by gagbo** [comment]
Probably fixed by #69 since this PR tries to stop ignoring timezones
